File: sample/java/src/main/java/com/bestseller/api/entity/Item.java
package com.bestseller.api.entity;

import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

import java.math.BigDecimal;
import java.time.Instant;

@Table("items")
public record Item(
    @Id
    @Column("item_id")
    String itemId,

    @Column("name")
    String name,

    @Column("price")
    BigDecimal price,

    @Column("description")
    String description,

    @Column("category")
    String category,

    @Column("brand")
    String brand,

    @Column("sku")
    String sku,

    @Column("created_at")
    Instant createdAt,

    @Column("updated_at")
    Instant updatedAt
) {
}

File: sample/dotnet/BestsellerApi.ApiService/Entities/Item.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BestsellerApi.ApiService.Entities;

[Table("items")]
public class Item
{
    [Key]
    [Column("item_id")]
    public string ItemId { get; set; } = string.Empty;

    [Required]
    [Column("name")]
    public string Name { get; set; } = string.Empty;

    [Required]
    [Column("price")]
    public decimal Price { get; set; }

    [Required]
    [Column("description")]
    public string Description { get; set; } = string.Empty;

    [Column("category")]
    public string? Category { get; set; }

    [Column("brand")]
    public string? Brand { get; set; }

    [Column("sku")]
    public string? Sku { get; set; }

    [Column("created_at")]
    public DateTimeOffset CreatedAt { get; set; }

    [Column("updated_at")]
    public DateTimeOffset UpdatedAt { get; set; }
}
